packages/Search/examples/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -12,21 +12,126 @@
 #include <DTK_DistributedSearchTree.hpp>
 
 #include <Kokkos_DefaultNode.hpp>
-#include <Teuchos_CommHelpers.hpp>
 #include <Teuchos_CommandLineProcessor.hpp>
-#include <Teuchos_DefaultComm.hpp>
-#include <Teuchos_StandardCatchMacros.hpp>
-#include <Teuchos_Time.hpp>
-#include <Teuchos_TimeMonitor.hpp>
 
+#include <mpi.h>
+
+#include <algorithm>
+#include <chrono>
 #include <cmath> // cbrt
+#include <numeric>
 #include <random>
+#include <utility>
+#include <vector>
+
+// Poor man's replacement for Teuchos::TimeMonitor
+class TimeMonitor
+{
+    using container_type = std::vector<std::pair<std::string, double>>;
+    using entry_reference_type = container_type::reference;
+    container_type _data;
+
+  public:
+    class Timer
+    {
+        entry_reference_type _entry;
+        bool _started;
+        std::chrono::high_resolution_clock::time_point _tick;
+
+      public:
+        Timer( entry_reference_type ref )
+            : _entry{ref}
+            , _started{false}
+        {
+        }
+        void start()
+        {
+            assert( !_started );
+            _tick = std::chrono::high_resolution_clock::now();
+            _started = true;
+        }
+        void stop()
+        {
+            assert( _started );
+            std::chrono::duration<double> duration =
+                std::chrono::high_resolution_clock::now() - _tick;
+            // NOTE I have put much thought into whether we should use the
+            // operator+= and keep track of how many times the timer was
+            // restarted.  To be honest I have not even looked was the original
+            // TimeMonitor behavior is :)
+            _entry.second = duration.count();
+            _started = false;
+        }
+    };
+    std::unique_ptr<Timer> getNewTimer( std::string name )
+    {
+        _data.emplace_back( std::move( name ), 0. );
+        return std::unique_ptr<Timer>( new Timer( _data.back() ) );
+    }
+    void summarize( MPI_Comm comm, std::ostream &os = std::cout )
+    {
+        // FIXME Haven't tried very hard to format the output.
+        int comm_size;
+        MPI_Comm_size( comm, &comm_size );
+        int comm_rank;
+        MPI_Comm_rank( comm, &comm_rank );
+        int n_timers = _data.size();
+        if ( comm_size == 1 )
+        {
+            os << "========================================\n\n";
+            os << "TimeMonitor results over 1 processor\n\n";
+            os << "Timer Name\tGlobal Time\n";
+            os << "----------------------------------------\n";
+            for ( int i = 0; i < n_timers; ++i )
+            {
+                os << _data[i].first << "\t" << _data[i].second << "\n";
+            }
+            os << "========================================\n";
+            return;
+        }
+        std::vector<double> all_entries( comm_size * n_timers );
+        std::transform( _data.begin(), _data.end(),
+                        &all_entries[comm_rank * n_timers],
```

Comment:
You are not guaranteed that `n_timers > 0`.

Reply:
Good point